pkg/controller/agent/fleet.go

```diff
@@ -265,6 +277,30 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
 	return token
 }
 
+func getKibanaBasePath(ctx context.Context, client k8s.Client, kibanaNSN types.NamespacedName) (string, error) {
+	var kb v1.Kibana
+	err := client.Get(ctx, kibanaNSN, &kb)
+	if err != nil {
+		return "", err
+	}
+
+	kibanaRawConfig := kb.Spec.Config
+	if kibanaRawConfig != nil {
```

Comment:
I am not sure if there is a better way to read this config, this seems a little complicated.

Reply:
I changed this, I think now it is a little better, thanks @naemono for the suggestion